dds/DCPS/AbstractSample.h

```diff
@@ -0,0 +1,262 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_ABSTRACT_SAMPLE_H
+#define OPENDDS_DCPS_ABSTRACT_SAMPLE_H
+
+#include "Serializer.h"
+#include "TypeSupportImpl.h"
+#include "RcHandle_T.h"
+#include "FilterEvaluator.h"
+#include "XTypes/DynamicDataAdapter.h"
+
+#include <dds/DdsDynamicDataC.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+class AbstractSample;
+typedef RcHandle<AbstractSample> AbstractSample_rch;
+
+/**
+ * Represents a sample that can either be an instance of a C++ generated type
+ * from opendds_idl or a DynamicData. This is meant to be used by
+ * DataReaderImpl and DataWriterImpl.
+ */
+class OpenDDS_Dcps_Export AbstractSample : public virtual RcObject {
+public:
+  AbstractSample(bool owns_data, bool read_only, bool key_only)
```

Comment:
Instead of bools, should it use a flags enum/integer?  It would be more self-documenting at the calling sites.

Reply:
It would, but the values aren't together in the templated Sample constructor. I can make separate enums. Also I realized Dynamic Sample doesn't need `owns_data`, so I'm moving that to the templated Sample.